### PR TITLE
chore: move to ruff-format

### PR DESCRIPTION
**Pull request by henryiii**
Moving to ruff-fmt. Faster, drops a dependency, less config, and formatting is a bit better.

See https://github.com/scientific-python/cookie/pull/299 and https://astral.sh/blog/the-ruff-formatter.

### CODE REVIEW

File: bin/update_pythons.py [hunk 1]
```diff
@@ -204,10 +204,8 @@ def update_version_macos(
         unsorted_versions = spec.filter(self.versions_dict)
         sorted_versions = sorted(unsorted_versions, reverse=True)
 
-        if version <= Version("3.8.9999"):
-            file_ident = "macosx10.9.pkg"
-        else:
-            file_ident = "macos11.pkg"
+        macver = "10.9" if version <= Version("3.8.9999") else "11"
```
Comment (henryiii) [inline bin/update_pythons.py:207]:
```suggestion
        macver = "x10.9" if version <= Version("3.8.9999") else "11"
```

File: bin/update_pythons.py [hunk 2]
```diff
@@ -204,10 +204,8 @@ def update_version_macos(
         unsorted_versions = spec.filter(self.versions_dict)
         sorted_versions = sorted(unsorted_versions, reverse=True)
 
-        if version <= Version("3.8.9999"):
-            file_ident = "macosx10.9.pkg"
-        else:
-            file_ident = "macos11.pkg"
+        macver = "x10.9" if version <= Version("3.8.9999") else "11"
+        file_ident = f"macosx{macver}.pkg"
```
Comment (henryiii) [inline bin/update_pythons.py:208]:
```suggestion
        file_ident = f"macos{macver}.pkg"
```

**Review by joerick — APPROVED** [review]
Some nice improvements, especially with multi line strings :)

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -13,16 +13,12 @@ repos:
   - id: mixed-line-ending
   - id: trailing-whitespace
 
-- repo: https://github.com/psf/black-pre-commit-mirror
-  rev: 23.10.1
-  hooks:
-  - id: black
-
 - repo: https://github.com/astral-sh/ruff-pre-commit
   rev: v0.1.2
```
Comment (Czaki) [inline .pre-commit-config.yaml:17]:
```suggestion
  rev: v0.1.3
```

It is released so maybe bump.

**Pull request merged**